File: Programowanie_Obiektowe/PO_C/PO_C/Owca.cpp
#include "Owca.h"

using namespace std;

Owca::Owca(int x, int y, Swiat& swiat) : Zwierze(OWCA_SILA, OWCA_INICJATYWA, x, y, swiat) {}

char Owca::rysowanie() const
{
    return OWCA;
}

Organizm* Owca::stworzPotomka(int x, int y) const
{
    return new Owca(x, y, swiat);
}

File: Programowanie_Obiektowe/PO_JAVA/src/Owca.java
public class Owca extends Zwierze
{
    public Owca(int x, int y, Swiat swiat)
    {
        super(Stale.OWCA_SILA, Stale.OWCA_INICJATYWA, x, y, swiat);
    }

    @Override
    public char rysowanie()
    {
        return Stale.OWCA;
    }

    @Override
    public String nazwa()
    {
        return Stale.OWCA_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Owca(x, y, swiat);
    }
}